internal/services/streamanalytics/stream_analytics_output_blob_resource.go

```diff
@@ -147,20 +156,28 @@ func resourceStreamAnalyticsOutputBlobCreateUpdate(d *pluginsdk.ResourceData, me
 				BlobOutputDataSourceProperties: &streamanalytics.BlobOutputDataSourceProperties{
 					StorageAccounts: &[]streamanalytics.StorageAccount{
 						{
-							AccountKey:  utils.String(storageAccountKey),
 							AccountName: utils.String(storageAccountName),
 						},
 					},
-					Container:   utils.String(containerName),
-					DateFormat:  utils.String(dateFormat),
-					PathPattern: utils.String(pathPattern),
-					TimeFormat:  utils.String(timeFormat),
+					Container:          utils.String(containerName),
+					DateFormat:         utils.String(dateFormat),
+					PathPattern:        utils.String(pathPattern),
+					TimeFormat:         utils.String(timeFormat),
+					AuthenticationMode: streamanalytics.AuthenticationMode(d.Get("authentication_mode").(string)),
 				},
 			},
 			Serialization: serialization,
 		},
 	}
 
+	if v, ok := d.GetOk("storage_account_key"); ok {
+		blobOutputDataSource, ok := props.Datasource.AsBlobOutputDataSource()
+		if !ok {
+			return fmt.Errorf("converting Output Data Source to a Blob Output: %+v", id)
+		}
+		(*blobOutputDataSource.StorageAccounts)[0].AccountKey = utils.String(v.(string))
```

Comment:
this isn't guaranteed, particularly with upcoming changes to the SDK - we'll want to set this back to `props.Datasource`

Reply:
Seems I cannot set it back to `props.Datasource` otherwise API would throw error.

```
Error: updating Output: (Name "acctestinput-220630081903063616" / Streamingjob Name "acctestjob-220630081903063616" / Resource Group "acctestRG-220630081903063616"): streamanalytics.OutputsClient#Update: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BadRequest" Message="The JSON provided in the request body is invalid. Property 'accountKey' value length 0 is not in the acceptable range from 2 to 4194304." Details=[{"code":"400","correlationId":"xxxxxxx","message":"The JSON provided in the request body is invalid. Property 'accountKey' value length 0 is not in the acceptable range from 2 to 4194304.","requestId":"xxxxxxx"}]
```

```
Error: creating Output: (Name "acctestinput-220630083216720683" / Streamingjob Name "acctestjob-220630083216720683" / Resource Group "acctestRG-220630083216720683"): streamanalytics.OutputsClient#CreateOrReplace: Failure responding to request: StatusCode=422 -- Original Error: autorest/azure: Service returned an error. Status=422 Code="422" Message="The required property 'accountKey' is missing from the request." Details=[{"code":"422","correlationId":"xxxxx","message":"The required property 'accountKey' is missing from the request.","requestId":"xxxxx"}]
```